File: spinwishapp/lib/models/auth/login_request.dart
class LoginRequest {
  final String emailAddress;
  final String password;

  LoginRequest({
    required this.emailAddress,
    required this.password,
  });

  Map<String, dynamic> toJson() => {
    'emailAddress': emailAddress,
    'password': password,
  };

  factory LoginRequest.fromJson(Map<String, dynamic> json) => LoginRequest(
    emailAddress: json['emailAddress'],
    password: json['password'],
  );
}

File: backend/src/main/java/com/spinwish/backend/models/requests/users/LoginRequest.java
package com.spinwish.backend.models.requests.users;

import com.spinwish.backend.validators.ValidEmail;
import jakarta.validation.constraints.NotBlank;
import lombok.Data;

@Data
public class LoginRequest {

    @NotBlank(message = "Email address is required")
    @ValidEmail
    private String emailAddress;

    @NotBlank(message = "Password is required")
    private String password;
}
